packages/unified-selection/src/unified-selection/iModel/IModel.ts

```diff
@@ -0,0 +1,136 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+/** @packageDocumentation
+ * @module UnifiedSelection
+ */
+
+/**
+ * Used as an argument to a function that can accept one or more [[Id64String]]s.
+ */
+export type IdArg = string | Set<string> | string[];
+
+/**
+ * Identifies the type of changes made to the [[SelectionSet]] to produce a [[SelectionSetEvent]].
+ */
+export enum SelectionSetEventType {
+  /** Elements have been added to the set. */
+  Add,
+  /** Elements have been removed from the set. */
+  Remove,
+  /** Some elements have been added to the set and others have been removed. */
+  Replace,
+  /** All elements are about to be removed from the set. */
+  Clear,
+}
+
+/**
+ * Passed to [[SelectionSet.onChanged]] event listeners when elements are added to the selection set.
+ */
+export interface SelectAddEvent {
+  /** The type of operation that produced this event. */
+  type: SelectionSetEventType.Add;
+  /** The Ids of the elements added to the set. */
+  added: IdArg;
+  /** The affected SelectionSet. */
+  set: SelectionSet;
+}
+
+/**
+ * Passed to [[SelectionSet.onChanged]] event listeners when elements are removed from the selection set.
+ */
+export interface SelectRemoveEvent {
+  /** The type of operation that produced this event. */
+  type: SelectionSetEventType.Remove | SelectionSetEventType.Clear;
+  /** The element Ids removed from the set. */
+  removed: IdArg;
+  /** The affected SelectionSet. */
+  set: SelectionSet;
+}
+
+/**
+ * Passed to [[SelectionSet.onChanged]] event listeners when elements are simultaneously added to and removed from the selection set.
+ */
+export interface SelectReplaceEvent {
+  /** The type of operation that produced this event. */
+  type: SelectionSetEventType.Replace;
+  /** The element Ids added to the set. */
+  added: IdArg;
+  /** The element Ids removed from the set. */
+  removed: IdArg;
+  /** The affected SelectionSet. */
+  set: SelectionSet;
+}
+
+/**
+ * Payload sent to [[SelectionSet.onChanged]] event listeners to describe how the contents of the set have changed.
+ */
+export type SelectionSetEvent = SelectAddEvent | SelectRemoveEvent | SelectReplaceEvent;
+
+/**
+ * Manages a set of *listeners* for a particular event and notifies them when the event is raised.
+ */
+export interface Event<T> {
```

Comment:
I think we should move existing `Event` interface in `core-interop` to `presentation-shared` and reuse it from there.

Reply:
Moved the interface into `presentation-shared` and reused it.